File: app/src/main/java/com/example/quarter/di/RepositoriesModule.java
package com.example.quarter.di;

import com.example.quarter.data.NotesRepositoryImpl;
import com.example.quarter.domain.NotesRepository;
import dagger.Binds;
import dagger.Module;

@Module
public interface RepositoriesModule {

    @Binds
    NotesRepository notesRepository(NotesRepositoryImpl repository);

}

File: app/src/main/java/com/example/quarter/di/RepositoriesModule.kt
package com.example.quarter.di

import com.example.quarter.data.NotesRepositoryImpl
import com.example.quarter.domain.NotesRepository
import dagger.Binds
import dagger.Module

@Module
interface RepositoriesModule {

    @Binds
    fun notesRepository(repository: NotesRepositoryImpl): NotesRepository
}
